scenarios/Makefile

```diff
@@ -77,10 +117,46 @@ stc-dev-build:
 	cd /tmp/onos-stc && mvn -q -B -Dmaven.test.skip=true install
 
 pull: ./tmp/.env.docker
+ifeq ($(ONOS_IMAGE),sdfabric-onos-local:master)
+	@#avoid pulling the ONOS image if we build it locally
+	${DOCKER_COMPOSE_CMD} pull ${STC_COMPOSE_SERVICES_WO_ONOS}
+else
 	${DOCKER_COMPOSE_CMD} pull
+endif
 
 build: ./tmp/.env.docker
+ifeq ($(ONOS_IMAGE),sdfabric-onos-local:master)
+	@#avoid pulling the ONOS image if we build it locally
+	${DOCKER_COMPOSE_CMD} build --pull ${STC_COMPOSE_SERVICES_WO_ONOS}
+else
 	${DOCKER_COMPOSE_CMD} build --pull
+endif
+
+build-sdfabric-onos: .build-sdfabric-onos
+.build-sdfabric-onos: ./tmp/.env.docker sdfabric-onos
+ifeq ($(ONOS_IMAGE),sdfabric-onos-local:master)
```

Comment:
what about if in this step we apply the local branch we are working on?
1) We have to create a branch
2) apply the diff with the master as patch
3) commit
4) use the commit name as UP4_VERSION

If we do this, we will not encounter any problem with the makefile. We don't need to do this now but it is good to remember

Reply:
I'd prefer if we keep the complexity of this Makefile under control. This can easily explode to the point where only a few of us can maintain it.